src/libtrx/game/lara/misc.c

```diff
@@ -1,10 +1,48 @@
 #include "game/lara/misc.h"
 
+#include "game/effects.h"
 #include "game/items.h"
 #include "game/lara/common.h"
 #include "game/lara/const.h"
+#include "game/random.h"
 #include "game/rooms.h"
 
+void Lara_TouchLava(void)
+{
+    ITEM *const lara_item = Lara_GetItem();
+    LARA_INFO *const lara_info = Lara_GetLaraInfo();
+    if (lara_item->hit_points < 0 || lara_info->water_status == LWS_CHEAT) {
+        return;
+    }
+
+    int16_t room_num = lara_item->room_num;
+    const SECTOR *const sector = Room_GetSector(
+        lara_item->pos.x, MAX_HEIGHT, lara_item->pos.z, &room_num);
+    const int32_t height =
+        Room_GetHeight(sector, lara_item->pos.x, MAX_HEIGHT, lara_item->pos.z);
+    if (lara_item->floor != height) {
+        return;
+    }
+
+    lara_item->hit_points = -1;
+    lara_item->hit_status = 1;
+
+    if (lara_info->water_status != LWS_ABOVE_WATER) {
+        return;
+    }
```

Comment:
Are there any underwater lava tiles in either of the OGs?

Reply:
In Diving Area there is the acid effect, and Temple of Xian has this too in many places (there are none in TR1). Actually, that's a good catch - this warrants a changelog entry. We added this check when introducing the ability to have lava tiles in water in TR1 to avoid flames showing for a frame (#1412). Pushed.